src/test/java/eu/xenit/gradle/docker/compose/DockerComposeConventionImplTest.java

```diff
@@ -228,5 +228,55 @@ public void testFromProjectWithDockerAlfrescoPlugin() {
         assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("SUB_PROJECT1_DOCKER_IMAGE"));
     }
 
+    @Test
+    public void testFromProjectWithEnvironmentVariable(){
+        composeConvention.fromProject("TESTNAME",project);
+        TaskCollection<DockerBuildImage> buildImageTaskCollection = project.getTasks().withType(DockerBuildImage.class);
+        assertTaskDependsOn(composeSettings.getUpTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getBuildTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getPushTask(), dockerBuildImageProvider);
+
+        runSetupImageEnvironmentWithComposeUp();
+
+        assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("TESTNAME_BUILD_DOCKER_IMAGE"));
+    }
 
+    @Test
+    public void testFromProjectWithEnvironmentVariable1(){
+        String DOCKER_IMAGE_ID_2 = "c0ff34";
+        TaskProvider<DockerBuildImage> dockerBuildImageProvider = project.getTasks()
+                .register("buildDockerImageAlternative", DockerBuildImage.class);
+        dockerBuildImageProvider.configure(dockerBuildImage -> {
+            dockerBuildImage.getImageId().set(DOCKER_IMAGE_ID_2);
+        });
+        composeConvention.fromProject("TESTNAME2",project);
```

Comment:
I notice you try to keep the names unique.

It does not hurt, but it is also not necessary. Each testcase gets their own isolated environment, so there are no name collisions between testcases.